src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -1089,6 +1110,31 @@ private long slowReadFlexUInt_1_1() {
         return slowReadLargeFlexUInt_1_1(currentByte);
     }
 
+    /**
+     * Reads the length of a FlexUInt (or FlexInt) at the given position.
+     * Does not alter the state of the peekIndex. May fill data, if needed.
+     * @return the number of bytes used to encode the FlexUInt (or FlexInt) that starts a "position"
+     *         or -1 if the end of the stream has been reached
+     */
+    private long slowReadLengthOfFlexUInt_1_1(long position) {
+        int length = 1;
+        while (true) {
+            if (!fillAt(position, 0)) {
```

Comment:
```suggestion
            if (!fillAt(position, 1)) {
```
Because we want to ensure at least one byte is available? To test for this, we feed bytes to the reader one-by-one and ensure it returns a value once the final byte is available. These tests have so far always gone in `IonReaderContinuableTopLevelBinaryTest`, though we agree that that test has gotten bloated.

Reply:
We want to make sure that the specific byte at `position` is available. From what I can tell, the second argument is the number of bytes _after_ position that need to be available. Is that correct?

Reply:
I think the JavaDoc on `fillAt` is technically wrong. It should say "at", not "after". Looking at the code for `fillAt`, it looks to me like providing `0` when `position` is at `limit` (which is exclusive) would not fill a byte, which is not what's intended by this code.